src/managers/k8s.py

```diff
@@ -15,13 +18,16 @@
 
 from literals import SECURITY_PROTOCOL_PORTS, AuthMap, AuthMechanism
 
-logger = logging.getLogger(__name__)
-
 # default logging from lightkube httpx requests is very noisy
 logging.getLogger("lightkube").disabled = True
 logging.getLogger("lightkube.core.client").disabled = True
 logging.getLogger("httpx").disabled = True
 logging.getLogger("httpcore").disabled = True
+logging.getLogger("lightkube").setLevel(logging.CRITICAL)
+logging.getLogger("httpx").setLevel(logging.CRITICAL)
+logging.getLogger("httpcore").setLevel(logging.CRITICAL)
+
+logger = logging.getLogger(__name__)
 
 
 class K8sManager:
```

Comment:
Thanks!
I agree it's a bit ugly, and I'm not *that* happy with the solution as it stands.
I also thought about a singleton, but rejected it as we want to be able to have multiple instances for different pod/units at the same time.

I really like the private-injected idea! I'm not sure we gain a huge amount yet though. I'll see what others say.

Reply:
Uhm, actually before reading Alex comment, I actually thought something similar. I was thinking of this pattern:

```
class Foo:
    # This could be even an attribute if needed in some contexts
    def bar(self, arg)
        return self._bar(arg, self._get_ttl_hash())
        
    @cache   
    def _bar(arg, ttl):
        ....
```

Is this what you mean @Batalex ? if so, I like it! Such that the public facing API does not know about this, and it is all internal logic (since clients should not be worrying about this)

Reply:
Yup, that's what I understood also. OK, I'll make that change :)

Reply:
Done in [00f1fa7](https://github.com/canonical/kafka-k8s-operator/pull/133/commits/00f1fa7b35df2d2151e7a2f2e13cf5fda27ae5d8)

Reply:
Love it!